client/packages/common/src/ui/layout/tables/material-react-table/tableState/useColumnSizing.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useState } from 'react';
+import {
+  MRT_ColumnSizingState,
+  MRT_RowData,
+  MRT_TableOptions,
+} from 'material-react-table';
+import { getSavedState, updateSavedState, differentOrUndefined } from './utils';
+import { useDebounceCallback } from '@common/hooks';
+
+export const useColumnSizing = (tableId: string) => {
+  const initial = { 'mrt-row-expand': 40 };
+
+  const [state, setState] = useState<MRT_ColumnSizingState>(
+    getSavedState(tableId).columnSizing ?? initial
+  );
+
+  const debouncedUpdateSavedState = useDebounceCallback(
+    (newState: MRT_ColumnSizingState) => {
+      updateSavedState(tableId, {
+        columnSizing: differentOrUndefined(newState, initial),
+      });
+    },
+    [],
+    500
```

Comment:
Using a debounced handler does mean that the first time you resize a column, the `reset table` button won't enable. It will enable after any other change to the table though (a second resize, or hide a column etc.)